### DISCUSSION TITLE
use rabbitmq_auth_backend_http and rabbitmq_auth_backend_cache，but still be queries within the cache time

### DISCUSSION THREAD

**Discussion by mthgh0818** [rabbitmq/rabbitmq-server#10276] [Questions] [answered]
I use RabbitMQ 3.12.10 for my mqtt broker , use  rabbitmq_auth_backend_cache and rabbitmq_auth_backend_http for user check ,。The specific configuration is as follows
```
[
  {rabbit, [{auth_backends, [rabbit_auth_backend_http, rabbit_auth_backend_internal,rabbit_auth_backend_cache]}]},
  {rabbitmq_auth_backend_cache,
     [{cache_module,rabbit_auth_cache_ets},
      {cache_module_args,[]},
      {cache_refusals,false},
      {cache_ttl,600000},
      {cached_backend,rabbit_auth_backend_internal}]},
  {rabbitmq_auth_backend_http,
   [{http_method,   get},
    {user_path,     "http://xxx.xxx.xxx.xxx:8080/auth/user"},
    {vhost_path,    "http://xxx.xxx.xxx.xxx:8080/auth/vhost"},
    {resource_path, "http://xxx.xxx.xxx.xxx:8080/auth/resource"},
    {topic_path,    "http://xxx.xxx.xxx.xxx:8080/auth/topic"}]}
].
```
Now I have encountered a problem. The same client publish message will still be verified within the cache time (60s), but it will not be verified every time, almost every 5s. Is it a problem with my configuration?
There is another question. When using rabbitmq_auth_backend_http, can I only use user_path without verifying resource_path and topic_path?

**Comment by michaelklishin** [comment, marked as answer]
@mthgh0818 if you use the caching backend, you must be OK with the fact that no external services will be consulted during the caching period.

5s sounds like the standard management UI refresh rate. Which uses user tags for certain parts of its permission tags.

You can point `resource_path` and `topic_path` to (your service's) HTTP endpoints that allow all operations or refuse all operations. You cannot partially disable what authZ operations a RabbitMQ node performs.

**Reply by mthgh0818** [reply]
Thank you for your reply. I have found the problem. It is a problem with my configuration
